include/devicetree/reserved.h

```diff
@@ -0,0 +1,141 @@
+/**
+ * @file
+ * @brief reserved-memory Devicetree macro public API header file.
+ */
+
+/*
+ * Copyright (c) 2021 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DEVICETREE_RESERVED_H_
+#define ZEPHYR_INCLUDE_DEVICETREE_RESERVED_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup devicetree-reserved Devicetree reserved-memory API
+ * @ingroup devicetree
+ * @{
+ */
+
+#include <linker/devicetree_reserved.h>
+
+/**
+ * @brief Get the pointer to the reserved-memory region
```

Comment:
rename macros in here to be `DT_RESERVED_MEMORY_` or some form of that `DT_RESERVED_MEM_`..

Reply:
ok